### DISCUSSION TITLE
ADB via Wifi needs to enabled everytime?

### DISCUSSION THREAD

**Discussion by Shamshala** [seaky/nspanel_pro_tools_apk#78] [Q&A]
Hello guys,

just a quick question because I wasn't able to confirm it. I was searching this related to general Android questions and it seems like it's normal for Android devices so I was wondering whether it's the case for you as well or there is a workaround. When you talk about ADB do you disassemble your NSPanel to get to the USB (alternatively cut the whole through chassis) everytime or is it supposed to be "one time only" and then the ADB server via network should just work? Because despite being able to enable it, I lost connection after few minutes and it needs to be connected to the PC directly again.

Thank you for any tip

**Comment by seaky** [comment]
When you enable adb you can connect also by usb or via wifi.

Of course adb over wifi is more convenient.

After certain period the adb disconnects the device which is normal, just reconnect it with adb connect command.

I have multiple devices and I have not opened them since I enabled adb.

**Reply by Shamshala** [reply]
That's the problem for me, when it disconnects and I try to connect again via command, it acts like there isn't any ADB running on the device.

But good to know. That means there is something wrong with my NSPanel, setup or network.

And should ADB work even after restart of a panel or do I need to set anything else for that?

Thank you!

**Reply by ShadOLurk** [reply]
In my experience, the only time I loose the ability  to adb via WiFi is if I launch the official panel app (ewelink).

The official app disables USB debug on launch, which in turn affects the Wi-Fi ADB.

A workaround I've implemented is to pair a small Bluetooth keyboard/pad to navigate the panel to re-enable debug mode.

**Reply by seaky** [reply]
"The official app disables USB debug on launch, which in turn affects the Wi-Fi ADB."
That is true when you have not rooted your device officially. If you used the mobile app and enabled developer mode there. The original apk will keep the device opened even if you start the ewelink app.
